src/overlays/actors/ovl_En_Estone/z_en_estone.c

```diff
@@ -31,33 +35,235 @@ const ActorInit En_Estone_InitVars = {
+        this->scale = (Rand_ZeroFloat(1.0f) * 0.003f) + 0.003f;
+        this->actor.velocity.y = Rand_ZeroFloat(5.0f) + 7.0f;
+        this->actor.gravity = -1.0f;
+    }
+    this->collider.dim.radius = this->scale * 3000.0f;
+    this->collider.dim.height = this->scale * 3000.0f;
+    this->collider.dim.yShift = this->scale * -1300.0f;
+    for (i = 0; i < 7; i++) {
+        accel.x = 0.5f * (Rand_ZeroOne() - 0.5f);
+        accel.y = -1.0f;
+        accel.z = 0.5f * (Rand_ZeroOne() - 0.5f);
+        vel.x = 2.0f * (Rand_ZeroOne() - 0.5f);
+        vel.y = 8.0f + (Rand_ZeroOne() * 10.0f);
+        vel.z = 2.0f * (Rand_ZeroOne() - 0.5f);
+        EnEstone_SpawnEffect(this, &this->actor.world.pos, &vel, &accel, (Rand_ZeroFloat(1.0f) * 0.001f) + 0.001f, 70);
+    }
+    this->timer = 5;
+    this->inactive = false;
+    Actor_SetScale(&this->actor, 1.0f);
+    this->actionFunc = EnEstone_Active;
+}
 
-extern UNK_TYPE D_06010240;
+void EnEstone_Destroy(Actor* thisx, PlayState* play) {
+    EnEstone* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Estone/EnEstone_Init.s")
+    Collider_DestroyCylinder(play, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Estone/EnEstone_Destroy.s")
+void EnEstone_Active(EnEstone* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    Vec3f accel;
+    Vec3f vel;
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Estone/func_80A9A1DC.s")
+    if ((this->actor.floorHeight == -10000.0f) || ((player->actor.world.pos.y - 200.0f) > this->actor.world.pos.y)) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    if (((this->actor.bgCheckFlags & 1) && (this->actor.velocity.y < 0.0f)) || (this->collider.base.atFlags & 4)) {
+        this->collider.base.atFlags &= ~4;
+        if (this->actor.params == ENESTONE_LARGE) {
+            for (i = 0; i < 2; i++) {
+                accel.x = 2.0f * (Rand_ZeroOne() - 0.5f);
+                accel.y = -1.0f;
+                accel.z = 2.0f * (Rand_ZeroOne() - 0.5f);
+                vel.x = 3.0f * (Rand_ZeroOne() - 0.5f);
+                vel.y = 5.0f + (Rand_ZeroOne() * 10.0f);
+                vel.z = 3.0f * (Rand_ZeroOne() - 0.5f);
+                EnEstone_SpawnEffect(this, &this->actor.world.pos, &vel, &accel,
+                                     0.003f + (0.002f * Rand_ZeroFloat(1.0f)), 20);
+            }
+        }
+        Math_Vec3f_Copy(&vel, &this->actor.world.pos);
+        vel.y = this->actor.floorHeight;
+        Actor_SpawnFloorDustRing(play, &this->actor, &vel, 0.0f, 10, 6.0f, 50, 30, true);
+        this->actor.velocity.y = this->actor.gravity = 0.0f;
+        this->actor.speedXZ *= 0.3f;
+        this->actor.shape.shadowScale = 0.0f;
+        this->inactive = true;
+        this->timer = 50;
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_WALL_BROKEN);
+        this->actionFunc = EnEstone_Inactive;
+    } else {
+    label: //! @fake
```

Comment:
```suggestion
    label: //! FAKE:
```
This is the marker we've been using for these